### PR TITLE
Support k8s 1.28

### PR DESCRIPTION
**Pull request by joshuatcasey**
See https://www.pivotaltracker.com/story/show/185846474 for details.

NOTE: As of 2023-08-20, this has several commits that have workarounds to support things that changed in K8s 1.28. Please review this PR commit-by-commit.

### CODE REVIEW

File: internal/supervisor/server/server.go [hunk 1]
```diff
@@ -613,6 +613,10 @@ func getAggregatedAPIServerConfig(
 	// `kubectl explain` to work for the Supervisor's aggregated API resources.
 	serverConfig.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(
 		supervisoropenapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(scheme))
+	// serverConfig.OpenAPIConfig.Info.InfoProps.Title = "Pinniped Supervisor"
```
Comment (benjaminapetersen) [inline internal/supervisor/server/server.go:616]: What does the comment indicate? Is this just for our ref?

File: internal/concierge/server/server.go [hunk 2]
```diff
@@ -228,6 +228,10 @@ func getAggregatedAPIServerConfig(
 	// `kubectl explain` to work for the Concierge's aggregated API resources.
 	serverConfig.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(
 		conciergeopenapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(scheme))
+	// serverConfig.OpenAPIConfig.Info.InfoProps.Title = "Pinniped Concierge"
```
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply internal/concierge/server/server.go:231]: Ah, so these are currently the default "Generic API Server" and we should uncomment these lines and update them to actually state the name of these servers "Pinniped Concierge" and "Pinniped Supervisor".

[the remark this answers is not in the document]
Comment (joshuatcasey) [reply internal/concierge/server/server.go:231]: https://github.com/vmware-tanzu/pinniped/issues/1645

[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply internal/concierge/server/server.go:231]: Issue #1645

File: internal/supervisor/server/server.go (hunk 1 above)
Comment (benjaminapetersen) [reply internal/supervisor/server/server.go:616]: Issue #1645

File: internal/concierge/impersonator/impersonator.go [hunk 3]
```diff
@@ -513,7 +514,7 @@ func newImpersonationReverseProxyFunc(restConfig *rest.Config) (func(*genericapi
 			}
 
 			ae := audit.AuditEventFrom(r.Context())
-			if ae == nil {
+			if ae == nil || reflect.DeepEqual(*ae, auditinternal.Event{}) {
```
Comment (benjaminapetersen) [inline internal/concierge/impersonator/impersonator.go:517]: Strange, so the library used to use pointers and does not?

File: internal/concierge/impersonator/impersonator_test.go [hunk 4]
```diff
@@ -1158,11 +1158,6 @@ func TestImpersonatorHTTPHandler(t *testing.T) {
 					Groups: testGroups,
 					Extra:  testExtra,
 				}, nil, "")
-				ctx := audit.WithAuditContext(req.Context())
```
Comment (benjaminapetersen) [inline internal/concierge/impersonator/impersonator_test.go:1161]: context no longer needed?

Comment (benjaminapetersen) [reply internal/concierge/impersonator/impersonator_test.go:1161]: Ah, context is built in the helper function. 👍

File: internal/concierge/impersonator/impersonator_test.go [hunk 5]
```diff
@@ -1932,13 +1927,11 @@ func newRequest(t *testing.T, h http.Header, userInfo user.Info, event *auditint
 		ctx = request.WithUser(ctx, userInfo)
 	}
 
-	ae := &auditinternal.Event{Level: auditinternal.LevelMetadata}
+	ctx = audit.WithAuditContext(ctx)
 	if event != nil {
-		ae = event
+		ac := audit.AuditContextFrom(ctx)
```
Comment (benjaminapetersen) [inline internal/concierge/impersonator/impersonator_test.go:1932]: So, if `event` is `nil`, nothing gets created, but this still works?  Though we said that `Events` are no longer pointers?

Comment (benjaminapetersen) [reply internal/concierge/impersonator/impersonator_test.go:1932]: But `ac.Event = *event` is a pointer, and seems to work?

**Review by benjaminapetersen — APPROVED** [review]
Given we consider:
- #1643 
- #1645 
before our next release I'm good with this!

File: internal/concierge/impersonator/impersonator_test.go (hunk 5 above)
Comment (joshuatcasey) [reply internal/concierge/impersonator/impersonator_test.go:1932]: The type for `ac` has changed in v0.28.0:

* https://github.com/kubernetes/apiserver/blob/v0.27.0/pkg/audit/context.go#L43
* https://github.com/kubernetes/apiserver/blob/v0.28.0/pkg/audit/context.go#L43

**Pull request merged**